etc/update.sh

```diff
@@ -14,25 +14,31 @@ cat /srv/www/www-mathe-oscar/data/webhook.secret >> .htaccess
 bundle config set --local path 'vendor/bundle'
 bundle install
 
+# use the venv defined in .venv
+if [ ! -d ".venv" ]; then
+    echo "venv directory not found, initializing..."
+    python3 -m venv .venv
+    echo "done!"
+fi
+echo "Activating venv..."
+source .venv/bin/activate
+
+# install requirements for the python scripts
+echo "Installing python pre requisites..."
+python3 -m pip install -r etc/requirements.txt
+
 # get tutorial status
-statusurl=$(curl -SsL --retry 5 "https://api.github.com/repos/oscar-system/TutorialTesterforOscar/actions/workflows/CI.yml/runs?per_page=1" | jq  '.workflow_runs[0].jobs_url' | sed 's/"//g')
-curl -SsL --retry 5 $statusurl | jq '.["jobs"][1:] | .[] | .name+":"+.conclusion' | sed 's/^.*\s.*\s.*\s//' | sed 's/):/: /' | sed s'/"$//' > _data/examples_status.yml
+echo "Getting tutorial status....."
+./etc/tutorial_status.py
+echo "Done!"
 # get tutorial last modified dates
-cd etc
-python3 update-dates.py
-cd ..
+echo "Getting tutorial last modified dates......."
+./etc/update-dates.py
+echo "Done!"
 # update version info
-curl -SsL https://api.github.com/repos/oscar-system/Oscar.jl/releases/latest > latest.json
-version=$(jq '.["name"]' latest.json | sed 's/v//')
-date=$(jq '.["created_at"]' latest.json | cut -c2-11)
-year=$(echo $date | tr '-' ' ' | awk '{print $1}')
-month=$(echo $date | tr '-' ' ' | awk '{print $2}')
-day=$(echo $date | tr '-' ' ' | awk '{print $3}')
-echo "version: $version" > _data/release.yml
-echo "year: \"$year\"" >> _data/release.yml
-echo "month: \"$month\"" >> _data/release.yml
-echo "day: \"$day\"" >> _data/release.yml
-echo "date: \"$year-$month-$day\"" >> _data/release.yml
-rm latest.json
+echo "Getting OSCAR version info......."
+./etc/update-latest-release.py
+echo "Done!"
 # run jekyll
+echo "Running jekyl......."
```

Comment:
```suggestion
echo "Running jekyll......."
```
Instead of all these manuals print statements it may be more effective to add `set -x` at the start of the script